test/index.js

```diff
@@ -349,56 +349,73 @@ describe('integration', function(){
-        var obj = { a: '48dc32b2', b: '48dc32b2' };
-        assert.deepEqual(integration.map(obj, 'b'), ['48dc32b2']);
+        var option = { a: '48dc32b2', b: '48dc32b2' };
+        assert.deepEqual(integration.map(option, 'b'), ['48dc32b2']);
       });
 
       it('should use to-no-case to match keys', function(){
-        var obj = { 'My Event': '7b4fe803', 'other event': '2107007a' };
-        assert.deepEqual(integration.map(obj, 'my_event'), ['7b4fe803']);
+        var option = { 'My Event': '7b4fe803', 'other event': '2107007a' };
+        assert.deepEqual(integration.map(option, 'my_event'), ['7b4fe803']);
       });
     });
 
-    describe('when .options.events is an array', function(){
-      it('should return an empty array if the array isnt a map', function(){
-        var obj = ['one', 'two'];
-        assert.deepEqual(integration.map(obj, 'one'), []);
+    describe('when the mapped option is type "array"', function(){
+      it('should map value when present in option array', function(){
+        var option = ['one', 'two'];
+        assert.deepEqual(integration.map(option, 'one'), ['one']);
       });
 
-      it('should return an empty array when the array is empty', function(){
-        var obj = [];
-        assert.deepEqual(integration.map(obj, 'wee'), []);
+      it('should return an empty array when option array is empty', function(){
+        var option = [];
+        assert.deepEqual(integration.map(option, 'wee'), []);
       });
+    });
 
+    describe('when the mapped option is type "mixed"', function(){
       it('should return an empty array on mismatch', function(){
-        var obj = [{ key: 'my event', value: '1121f10f' }];
-        assert([], integration.map(obj, 'event'));
+        var option = [{ key: 'my event', value: '1121f10f' }];
+        assert([], integration.map(option, 'event'));
       });
 
-      it('should return all matches in the array', function(){
-        var obj = [{ key: 'baz', value: '4cff6219' }, { key: 'baz', value: '4426d54'} ];
-        assert(['4cff6219', '4426d54'], integration.map(obj, 'baz'));
+      it('should return an empty array on malformed options', function(){
+        var option = [{ key: 'my event', notValue: '1121f10f' }];
+        assert([], integration.map(option, 'event'));
       });
 
-      it('should return an array with the object on match when handling `mixed` values', function(){
-        var events = [
-          { key: 'testEvent', value: { event: 'testEvent', mtAdId: 'mt-ad-id', mtId: 'mt-id' } },
-          { key: 'testEvent2', value: { event: 'testEvent2', mtAdId: 'mt-ad-id', mtId: 'mt-id' } }
-        ];
-        assert.deepEqual(integration.map(events, 'testEvent'), [{ event: 'testEvent', mtAdId: 'mt-ad-id', mtId: 'mt-id' }]);
+      it('should return an empty array for non-string key', function(){
+        var option = [{ key: null, notValue: '1121f10f' }];
+        assert([], integration.map(option, 'event'));
```

Comment:
We're actually testing three distinct behaviors here: `key` is a non-string value, `value` doesn't exist, and the provided `key` (`'event'`) doesn't match any possible value in `option`, even if we were to coerce `null` to a string. This test would be better if it tested only one of those conditions: pass `null` as the second argument to `#map`. (Better, add a second `assert` where `null` is quoted to be sure there's no coercion business going on.

- `key` 

A better test case here would be to pass the actual value `null`, because we know that even if